resources/js/provider.js

```
@@ -152,6 +159,25 @@
     EthereumProvider.prototype.status = new StatusAPI();
     EthereumProvider.prototype.isConnected = function () { return true; };
 
+    EthereumProvider.prototype._events = {};
+
+    EthereumProvider.prototype.on = function(name, listener) {
+        if (!this._events[name]) {
+          this._events[name] = [];
+        }
+        this._events[name].push(listener);
+    }
+
+    EthereumProvider.prototype.removeListener = function (name, listenerToRemove) {
+        if (!this._events[name]) throw new Error(`event "${name}" does not exist`);
```

why do we want to throw an error here?